### DISCUSSION TITLE
Protip: Don't be afraid to run the script multiple times

### DISCUSSION THREAD

**Discussion by Madwonk** [danielewood/sierra-wireless-modems#71] [General]
Hey, first off, hats off and thanks to the project maintainer. This thing just saved me many hours of my weekend!

Just wanted to say that the main script *seems* to work perfectly fine under Debian 11 on my machine (though I'm guessing this isn't an officially supported configuration). However, it did take several runs to get things to properly flash- I initially got stuck at "Waiting for modem in boothold mode..." but after several re-runs (and flipping in/out of suspend a few times to see if that would help) it ended up flashing perfectly fine!

**Comment by Rudder2** [comment]
Thank you!!!  I Hibernated my computer and woke it up and it worked!!  Your message helped.  When I read you message I remembered my computer wakes up with out any custom Kernel Mods, annoying because my PC operates weird with out them but I wonder is this is why the Suspend / Hibernate fixes the script.  THANK YOU!!

**Comment by luckylinux** [comment]
I am facing similar issues.
Two things seemed to help in my case, although it's not 100% repeatable and you might need to disconnect-reconnect (physically) your USB adapter a few times until it works.

1. Use a USB 2.0 Port (using a USB 3.0 Port only worked the first flash, afterwards it was never detected)
2. Change the sleep "interval" for the while loop to something smaller than the currently used value of 2s.
    I used 0.1s instead of the default of 2s.
```
deviceid=''
while [ -z $deviceid ]
do
     echo 'Waiting for modem in boothold mode...'
     sleep 0.1
     deviceid=$(lsusb | grep -i -E '1199:9070|1199:9078|413C:81B5' | awk '{print $6}')
done
echo "Found $deviceid"
```

I believe the issue is that the boot process is too fast, then the "1199:9070" no longer exist by that time. Afterwards only "1199:9071" will exist.

Similar issue for "1199:9078" only existing in the early stages of the boot process. Afterwards only "1199:9079" will exist.

Thus the deviceid / $deviceid is always empy, unless you "catch" the right moment of time where the correct USB VID/PID appears during the "scan".
